### PR TITLE
Update build tools

### PR DESCRIPTION
**Pull request by serj-lotutovici**
androidBuildTools 23.0.2 -> 23.0.3
androidGradlePlugin 2.0.0 -> 2.1.2

### CODE REVIEW

**Comment by artem-zinnatullin** [comment]
Hi! You're not the only one: #172, #182. But I'd really like to merge that change.

Except Gradle Wrapper 2.14, it has pretty critical bug with non-clean builds: 

* https://code.google.com/p/android/issues/detail?id=213413
* https://youtrack.jetbrains.com/issue/KT-12750
* https://issues.gradle.org/browse/GRADLE-3491

**Comment by serj-lotutovici** [comment]
Ups didn't notice that... sorry.
> Hi! You're not the only one: #172, #182. But I'd really like to merge that change.

Will revert that (thx, for pointing that out)
> Except Gradle Wrapper 2.14, it has pretty critical bug with non-clean builds:

**Comment by serj-lotutovici** [comment]
@artem-zinnatullin `:app:testDebugUnitTest` is failing unexpectedly... any ideas why? same is happening in my second pr

**Comment by artem-zinnatullin** [comment]
I've restarted Travis build, Travis makes our Emulator tests flaky sometimes

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Enlarge the gradle heap size.
+org.gradle.jvmargs=-Xmx4096m -XX:MaxPermSize=1024m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
```
Comment (serj-lotutovici) [inline gradle.properties]: @artem-zinnatullin with this, the ci seams to have enough memory... ¯\_(ツ)_/¯

Comment (artem-zinnatullin) [inline gradle.properties]: Huh, ok then!

Can you please remove HeapDump flag? Don't think we need it

Comment (serj-lotutovici) [inline gradle.properties]: Done
> Can you please remove HeapDump flag? Don't think we need it

**Comment by vanniktech** [comment]
👍

**Comment by artem-zinnatullin** [comment]
Failed on CircleCI

> Your build has exceeded the memory limit of 4G on 1 container.

Can you try to allocate something like 3.5G?

**Comment by serj-lotutovici** [comment]
Green!

**Comment by artem-zinnatullin** [comment]
Merge!

**Comment by artem-zinnatullin** [comment]
Thanks 🐱

**Pull request merged**